File: stripe/src/test/java/com/stripe/android/FakeLogger.java
package com.stripe.android;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

final class FakeLogger implements Logger {
    @Override
    public void error(@NotNull String msg, @Nullable Throwable t) {

    }

    @Override
    public void info(@NotNull String msg) {

    }
}

File: stripe/src/test/java/com/stripe/android/FakeLogger.kt
package com.stripe.android

internal class FakeLogger : Logger {
    val debugLogs = mutableListOf<String>()
    val infoLogs = mutableListOf<String>()
    val errorLogs = mutableListOf<Pair<String, Throwable?>>()

    override fun debug(msg: String) {
        debugLogs.add(msg)
    }

    override fun error(msg: String, t: Throwable?) {
        errorLogs.add(msg to t)
    }

    override fun info(msg: String) {
        infoLogs.add(msg)
    }
}
